core/signals_test.go

```diff
@@ -63,42 +62,49 @@ func TestMaintenanceSignal(t *testing.T) {
 // because they'll interfere with each other's state.
 func TestTerminateSignal(t *testing.T) {
 
-	config := getSignalTestConfig()
+	app := getSignalTestConfig()
 	startTime := time.Now()
 	go func() {
-		if exitCode, _ := utils.ExecuteAndWait(config.Command); exitCode != 2 {
+		if exitCode, _ := utils.ExecuteAndWait(app.Command); exitCode != 2 {
 			t.Fatalf("Expected exit code 2 but got %d", exitCode)
 		}
 	}()
 	// we need time for the forked process to start up and this is async
 	runtime.Gosched()
 	time.Sleep(10 * time.Millisecond)
 
-	terminate(config)
+	app.Terminate()
 	elapsed := time.Since(startTime)
-	if elapsed.Seconds() > float64(config.StopTimeout) {
+	if elapsed.Seconds() > float64(app.StopTimeout) {
 		t.Fatalf("Expected elapsed time <= %d seconds, but was %.2f",
-			config.StopTimeout, elapsed.Seconds())
+			app.StopTimeout, elapsed.Seconds())
 	}
 }
 
 // Test handler for SIGHUP
 func TestReloadSignal(t *testing.T) {
-	oldConfig := getSignalTestConfig()
-	oldConfig.ConfigFlag = "invalid"
-	if badConfig := reloadConfig(oldConfig); badConfig != nil {
-		t.Errorf("Invalid configuration did not return nil")
+	app := getSignalTestConfig()
+	app.ConfigFlag = "invalid"
+	err := app.Reload()
+	if err == nil {
+		t.Errorf("Invalid configuration did not return error")
+	}
+	app.ConfigFlag = `{ "consul": "newconsul:8500" }`
+	err = app.Reload()
+	if err != nil {
+		t.Errorf("Valid configuration returned error: %v", err)
 	}
-	oldConfig.ConfigFlag = `{ "consul": "newconsul:8500" }`
-	if newConfig := reloadConfig(oldConfig); newConfig == nil || newConfig.Consul != "newconsul:8500" {
+	discSvc := app.DiscoveryService
+	if discSvc == nil {
```

Comment:
Shouldn't we also check for `NoopDiscoveryService`, rather than just that it was cleared to be `nil`?